src/main/java/com/rylinaux/plugman/util/BukkitCommandWrap.java

```diff
@@ -25,139 +31,81 @@ public class BukkitCommandWrap {
     private Constructor bukkitcommandWrapperConstructor;
 
     public BukkitCommandWrap() {
-        try {
-            this.nmsVersion = Bukkit.getServer().getClass().getPackage().getName().replace(".", ",").split(",")[3];
-        } catch (ArrayIndexOutOfBoundsException e) {
-            e.printStackTrace();
-            this.nmsVersion = null;
+        String prefix = cbPrefix();
+        String[] packageParts = prefix.split("\\.");
+        String versionPart = packageParts[packageParts.length - 1];
+        this.craftBukkitVersioning = versionPart.startsWith("v1_");
+        if (this.craftBukkitVersioning) {
+            this.nmsVersion = versionPart;
         }
+
+        boolean nmsVers = true;
+        try {
+            Class.forName("net.minecraft.server.MinecraftServer");
+            nmsVers = false;
+        } catch (ClassNotFoundException ignore) {}
+        this.nmsVersioning = nmsVers;
     }
 
-    public void wrap(Command command, String alias) {
-        if (this.nmsVersion == null) return;
-        if (this.minecraftServerClass == null) try {
-            this.minecraftServerClass = Class.forName("net.minecraft.server." + this.nmsVersion + ".MinecraftServer");
-        } catch (ClassNotFoundException e) {
-            try {
-                this.minecraftServerClass = Class.forName("net.minecraft.server.MinecraftServer");
-            } catch (ClassNotFoundException classNotFoundException) {
-                classNotFoundException.addSuppressed(e);
-                classNotFoundException.printStackTrace();
-                return;
-            }
-        }
+    private static @NotNull String cbPrefix() {
+        return Bukkit.getServer().getClass().getPackage().getName();
```

Comment:
I think you forgot something here...

What is the `craftBukkitVersioning` variable for, if we're ignoring it?

Reply:
I don't see what was forgotten. craftBukkitVersioning was for your use if you needed it later but that seems like it was unnecessary so I removed it

Reply:
> I don't see what was forgotten. craftBukkitVersioning was for your use if you needed it later but that seems like it was unnecessary so I removed it

Oh... Yeah, I misunderstood the code